File: android/app/src/main/java/com/smartexpense/android/domain/usecase/expense/GetExpensesUseCase.java
package com.smartexpense.android.domain.usecase.expense;

import com.smartexpense.android.data.remote.dto.response.ExpenseResponseDto;
import com.smartexpense.android.domain.repository.ExpenseRepository;
import com.smartexpense.android.domain.repository.ResultCallback;

import java.util.List;

public class GetExpensesUseCase {
    private final ExpenseRepository repository;

    public GetExpensesUseCase(ExpenseRepository repository) {
        this.repository = repository;
    }

    public void execute(ResultCallback<List<ExpenseResponseDto>> callback) {
        repository.getExpenses(callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/expense/GetExpensesUseCase.kt
package com.smartexpense.android.domain.usecase.expense

import com.smartexpense.android.data.remote.dto.response.ExpenseResponseDto
import com.smartexpense.android.domain.repository.ExpenseRepository

class GetExpensesUseCase(private val repository: ExpenseRepository) {
    suspend fun execute(): Result<List<ExpenseResponseDto>> {
        return repository.getExpenses()
    }
}
